backend/app/repositories/user_repo.py

```
@@ -0,0 +1,35 @@
+            last_name=last_name,
+            username=username,
+            password=password,
+            email=email,
+            phone_number=phone_number,
+        )
+        self.db.add(user)
+        try:
+            self.db.commit()
+        except IntegrityError:
+            self.db.rollback()
+            raise DuplicateUserError("email or username (or phone number) already exists")
```

Ruff (0.13.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)